### PR TITLE
quiche: build GSO quic writer

### PR DESCRIPTION
**Pull request by danzh2010**
Signed-off-by: Dan Zhang <danzh@google.com>

Build and test quic_gso_batch_writer under linux.

Added implementation for quic_udp_socket_platform_api.h

Risk Level: low, not in use
Testing: added quic_core_batch_writer_batch_writer_test.cc

### CODE REVIEW

**Comment by mattklein123** [comment]
Can you check tidy?

/wait

File: bazel/external/quiche.BUILD [hunk 1]
```diff
@@ -3741,3 +3939,17 @@ envoy_cc_test(
         ":quic_platform_test_mem_slice_vector_lib",
     ],
 )
+
+envoy_cc_test(
+    name = "quic_core_batch_writer_batch_writer_test",
+    srcs = ["quiche/quic/core/batch_writer/quic_batch_writer_test.cc"],
```
[the remark this answers is not in the document]
Comment (danzh2010) [reply bazel/external/quiche.BUILD]: good point! It should need the condition. Fixed.

**Comment by danzh2010** [comment]
Fixed clang-tidy and added quic_core_batch_writer_batch_writer_test to CI tests. PTAL

**Pull request merged**